src/nvim/terminal.c

```diff
@@ -353,8 +353,14 @@ void terminal_resize(Terminal *term, uint16_t width, uint16_t height)
   invalidate_terminal(term, -1, -1);
 }
 
-void terminal_enter(Terminal *term, bool process_deferred)
+void terminal_enter(bool process_deferred)
 {
+  Terminal *term = curbuf->terminal;
+  assert(term && "should only be called when curbuf has a terminal");
```

Comment:
What if you pass the buffer to `terminal_enter` instead of relying on the `curbuf` global? You can move the `assert` up on the call chain and handle the error (or totally avoid it) in a better way.

I see that the terminal also has a reference to the buffer. Maybe this is a sign that most the function should actually take the buffer as argument. @tarruda may have his reasons.